File: ccc17s3.py
import sys
input = sys.stdin.readline

num_blocks = int(input())
blocks = sorted(list(map(int, input().split())))

configs = []
heights = [0 for _ in range(blocks[-1] + blocks[-2])]

for a in range(len(blocks)):
	for b in range(a + 1, len(blocks)):
		if (blocks[a], blocks[b]) not in configs:
			heights[blocks[a] + blocks[b] - 1] += 1
			configs.append((blocks[a], blocks[b]))

max_width = max(heights)

print(max_width, heights.count(max(heights)))

File: ccc17s3.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int numBlocks;
    cin >> numBlocks;
    int blocks[numBlocks] = {};

    for (int i = 0; i < numBlocks; i++) {
        cin >> blocks[i];
    }

    sort(blocks, blocks + sizeof(blocks)/sizeof(blocks[0]));

    int maxWidth = 0;
    int maxCount = 0;

    for (int a = 0; a < numBlocks; a++) {
        for (int b = 0; b < a; b++) {
	    if (blocks[a] + blocks[b] > maxWidth) {
	        maxWidth = blocks[a] + blocks[b];
		maxCount = 1;
	    }
	    else if (blocks[a] + blocks[a] == maxWidth) {
		maxCount++;
	    }
	}
    }

    cout << maxWidth << " " << maxCount << endl;
}
